test/hotspot/gtest/riscv/test_assembler_riscv.cpp

```diff
@@ -234,82 +264,72 @@ void run_plain_cmpxchg_tests() {
                                            0 /* return */  , 1337 /* end value*/, true /* boolean ret*/);
 }
 
-TEST_VM(RiscV, cmpxchg_int64_plain_lr_sc) {
+TEST_VM(RiscV, cmpxchg_int64_lr_sc) {
   bool zacas = UseZacas;
   UseZacas = false;
   run_plain_cmpxchg_tests<int64_t, Assembler::int64>();
   UseZacas = zacas;
 }
 
-TEST_VM(RiscV, cmpxchg_int64_plain_maybe_zacas) {
+TEST_VM(RiscV, cmpxchg_int64_maybe_zacas) {
   if (UseZacas) {
     run_plain_cmpxchg_tests<int64_t, Assembler::int64>();
   }
 }
 
-TEST_VM(RiscV, cmpxchg_int32_plain_lr_sc) {
+TEST_VM(RiscV, cmpxchg_int32_lr_sc) {
   bool zacas = UseZacas;
   UseZacas = false;
   run_plain_cmpxchg_tests<int32_t, Assembler::int32>();
   UseZacas = zacas;
 }
 
-TEST_VM(RiscV, cmpxchg_int32_plain_maybe_zacas) {
+TEST_VM(RiscV, cmpxchg_int32_maybe_zacas) {
   if (UseZacas) {
     run_plain_cmpxchg_tests<int32_t, Assembler::int32>();
   }
 }
 
-template <typename TESTSIZE, Assembler::operand_size ASMSIZE>
-class NarrowCmpxchgTester {
- public:
-  typedef TESTSIZE (*cmpxchg_func)(intptr_t addr, TESTSIZE expected, TESTSIZE new_value, TESTSIZE result,
-                                   int64_t scratch0, int64_t scratch1, int64_t scratch2);
+TEST_VM(RiscV, cmpxchg_uint32_lr_sc) {
+  bool zacas = UseZacas;
+  UseZacas = false;
+  run_plain_cmpxchg_tests<uint32_t, Assembler::uint32>();
```

Comment:
Not quite sure if we should cover `uint32` type here. `lr.w` used by the CAS operation loads a 32-bit word and sign-extend it.

Reply:
As passing uint32 as operand size is supported and done by compareAndSwapNNode.
So we should test it ?

Which also says we should add test with max and min size of TESTSIZE, fixing that.

Reply:
So one case is shenandoah CAS:ing a narrow oop.
Narrow oop are zero extended (MASM::set_narrow_oop) which means expected, exchange values should be zero extended. Hence I think we have yet another cmpxchg bug.

Reply:
Sorry, I withdraw my previous comment. I think we are fine here.
I double checked and found that the the zero-extension is there in `MacroAssembler::load_reserved` for `uint32` type [1].

[1] https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/riscv/macroAssembler_riscv.cpp#L3693

Reply:
I found an issue in the gtest using uint32_t.